gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/domain_service/api/ValidateApiDomainServiceLegacyWrapper.java

```diff
@@ -16,35 +16,55 @@
 import io.gravitee.apim.core.api.model.Api;
+import io.gravitee.apim.core.exception.ValidationDomainException;
 import io.gravitee.apim.core.flow.domain_service.FlowValidationDomainService;
 import io.gravitee.apim.core.membership.model.PrimaryOwnerEntity;
 import io.gravitee.apim.infra.adapter.ApiAdapter;
 import io.gravitee.apim.infra.adapter.PrimaryOwnerAdapter;
+import io.gravitee.definition.model.DefinitionVersion;
+import io.gravitee.definition.model.v4.ApiType;
 import io.gravitee.rest.api.service.common.ExecutionContext;
 import io.gravitee.rest.api.service.v4.validation.ApiValidationService;
+import io.gravitee.rest.api.service.v4.validation.EndpointGroupsValidationService;
+import io.gravitee.rest.api.service.v4.validation.ListenerValidationService;
+import io.gravitee.rest.api.service.v4.validation.TagsValidationService;
+import java.util.Set;
+import lombok.AllArgsConstructor;
 import org.springframework.stereotype.Service;
 
+@AllArgsConstructor
 @Service
 public class ValidateApiDomainServiceLegacyWrapper implements ValidateApiDomainService {
 
     private final ApiValidationService apiValidationService;
     private final CategoryDomainService categoryDomainService;
     private final FlowValidationDomainService flowValidationDomainService;
-
-    public ValidateApiDomainServiceLegacyWrapper(
-        ApiValidationService apiValidationService,
-        CategoryDomainService categoryDomainService,
-        FlowValidationDomainService flowValidationDomainService
-    ) {
-        this.apiValidationService = apiValidationService;
-        this.categoryDomainService = categoryDomainService;
-        this.flowValidationDomainService = flowValidationDomainService;
-    }
+    private final TagsValidationService tagsValidationService;
+    private final GroupValidationService groupValidationService;
+    private final ListenerValidationService listenerValidationService;
+    private final EndpointGroupsValidationService endpointGroupsValidationService;
 
     @Override
     public Api validateAndSanitizeForCreation(Api api, PrimaryOwnerEntity primaryOwner, String environmentId, String organizationId) {
+        if (api.getDefinitionVersion() != DefinitionVersion.V4) {
+            throw new ValidationDomainException("Definition not supported, should be V4");
+        }
+
+        if (api.getType() == ApiType.NATIVE) {
+            this.validateAndSanitizeNativeV4(api, primaryOwner, environmentId, organizationId);
+        } else {
+            this.validateAndSanitizeHttpV4(api, primaryOwner, environmentId, organizationId);
+        }
+
+        api.setCategories(categoryDomainService.toCategoryId(api, environmentId));
+
+        return api;
+    }
+
+    private void validateAndSanitizeHttpV4(final Api api, PrimaryOwnerEntity primaryOwner, String environmentId, String organizationId) {
         var newApiEntity = ApiAdapter.INSTANCE.toNewApiEntity(api);
 
         apiValidationService.validateAndSanitizeNewApi(
```

Comment:
Check if this method (or part of it) should be executed also for Native API (look at `description` mainly)

Reply:
The steps that are in this method are executed in `validateAndSanitizeNativeV4` in `ValidateApiDomainServiceLegacyWrapper`. I didn't notice that it sanitizes the `description` however 🙈 . I will add that to `ValidateApiDomainServiceLegacyWrapper`.